template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/ConfigUtils.java

```diff
@@ -51,4 +52,15 @@ public static ApiClusterTemplateVariable variable(String name, String value) {
                 .findFirst();
     }
 
-}
+    public static Optional<StorageLocationView> getStorageLocationForServiceProperty(TemplatePreparationObject source, String serviceProperty) {
+        return source.getFileSystemConfigurationView()
+                .get().getLocations().stream()
+                .filter(s -> s.getProperty().equalsIgnoreCase(serviceProperty))
```

Comment:
This will definitely have to change; CM needs paths associated with a (service_name, role_name, template_parameter_key) tuple.

Reply:
CM accepts cloud storage location only as safety valve. The code above reads the user defined storage locations which is available as service-key value pairs in CB StorageLocationView.

Reply:
Yep, got it.